slack_sdk/socket_mode/aiohttp/__init__.py

```diff
@@ -126,70 +136,116 @@ def __init__(
         self.message_processor = asyncio.ensure_future(self.process_messages())
 
     async def monitor_current_session(self) -> None:
-        while not self.closed:
-            await asyncio.sleep(self.ping_interval)
-            try:
-                if self.auto_reconnect_enabled and (
-                    self.current_session is None or self.current_session.closed
-                ):
-                    self.logger.info(
-                        "The session seems to be already closed. Going to reconnect..."
+        try:
+            while not self.closed:
+                try:
+                    await asyncio.sleep(self.ping_interval)
+                    if self.current_session is not None:
+                        t = time.time()
+                        if self.last_ping_pong_time is None:
+                            self.last_ping_pong_time = float(t)
+                        await self.current_session.ping(f"ping-pong:{t}")
+
+                    if self.auto_reconnect_enabled:
+                        should_reconnect = False
+                        if self.current_session is None or self.current_session.closed:
+                            self.logger.info(
+                                "The session seems to be already closed. Going to reconnect..."
+                            )
+                            should_reconnect = True
+
+                        if self.last_ping_pong_time is not None:
+                            disconnected_seconds = int(time.time() - self.last_ping_pong_time)
+                            if disconnected_seconds >= (self.ping_interval * 4):
+                                self.logger.info(
+                                    "The connection seems to be stale. Disconnecting..."
+                                    f" reason: disconnected for {disconnected_seconds}+ seconds)"
+                                )
+                                self.stale = True
+                                self.last_ping_pong_time = None
+                                should_reconnect = True
+
+                        if should_reconnect is True or not await self.is_connected():
+                            await self.connect_to_new_endpoint()
+                            self.logger.info("Reconnection done.")
```

Comment:
Suggest `"Reconnected to endpoint"` (with no period at end; other log messages don't end with punctuation)